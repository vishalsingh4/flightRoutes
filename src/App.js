import React from 'react';

// components import
import FlightDetails from 'components/FlightDetails';
import HeaderComponent from 'components/HeaderComponent';
import FooterComponent from 'components/FooterComponent';

function App() {
  return (
    <div className="app">
      <header className='header-section'>
        <HeaderComponent />
      </header>
      <main className='main-section'>
        <FlightDetails />
      </main>
      <footer className='footer-section'>
        <FooterComponent />
      </footer>
    </div>
  );
}

export default App;
