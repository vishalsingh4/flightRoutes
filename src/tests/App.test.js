import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe("App Component", () => {
    const wrapper = shallow(<App />);
    it('should render the Header Component correctly - snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('renders component', () => {
        expect(wrapper.exists()).toBe(true);
    });
    it('renders the HeaderComponent component', () => {
        expect(wrapper.find('HeaderComponent').exists()).toBe(true);
    });
    it('renders the FlightDetails component', () => {
        expect(wrapper.find('FlightDetails').exists()).toBe(true);
    });
    it('renders the FooterComponent component', () => {
        expect(wrapper.find('FooterComponent').exists()).toBe(true);
    });
})