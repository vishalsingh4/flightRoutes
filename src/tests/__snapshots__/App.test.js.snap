// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render the Header Component correctly - snapshot 1`] = `
<div
  className="app"
>
  <header
    className="header-section"
  >
    <HeaderComponent />
  </header>
  <main
    className="main-section"
  >
    <FlightDetails />
  </main>
  <footer
    className="footer-section"
  >
    <FooterComponent />
  </footer>
</div>
`;
